src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,62 @@
+package roomescape.service;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import roomescape.dao.ReservationDao;
+import roomescape.dao.ReservationTimeDao;
+import roomescape.domain.reservation.Reservation;
+import roomescape.domain.reservationtime.ReservationTime;
+import roomescape.dto.reservation.ReservationCreateRequest;
+import roomescape.dto.reservation.ReservationResponse;
+
+@Service
+public class ReservationService {
+
+    private final ReservationDao reservationDao;
+    private final ReservationTimeDao reservationTimeDao;
+
+    public ReservationService(ReservationDao reservationDao, ReservationTimeDao reservationTimeDao) {
+        this.reservationDao = reservationDao;
+        this.reservationTimeDao = reservationTimeDao;
+    }
+
+    public List<ReservationResponse> findAll() {
+        List<Reservation> reservations = reservationDao.findAll();
+        return reservations.stream()
+                .map(ReservationResponse::from)
+                .toList();
+    }
+
+    public ReservationResponse add(ReservationCreateRequest request) {
+        validateNotExistReservationTime(request.getTimeId());
+        ReservationTime reservationTime = reservationTimeDao.findById(request.getTimeId());
+        Reservation reservation = request.toDomain(reservationTime);
+        long id = reservationDao.add(reservation);
+        Reservation result = reservationDao.findById(id);
+        return ReservationResponse.from(result);
+    }
```

Comment:
도메인을 `add` 할 때 아래와 같은 방식을 시도해볼 수도 있어요.
DB 쿼리를 한 번 더 날리지 않게 하는거죠.
```suggestion
    public ReservationResponse add(ReservationCreateRequest request) {
        validateNotExistReservationTime(request.getTimeId());
        ReservationTime reservationTime = reservationTimeDao.findById(request.getTimeId());
        Reservation reservation = request.toDomain(reservationTime);
        
        Reservation result = reservationDao.add(reservation);
        return ReservationResponse.from(result);
    }
```
물론 다온의 방식으로 하는 경우도 있기 때문에 꼭 바꾸지 않으셔도 괜찮아요.
어떤 생각을 갖고 이렇게 만드셨는지가 궁금합니다!

Reply:
```java
public ReservationTime add(ReservationTime reservationTime) {
    String sql = """
            INSERT
            INTO reservation_time
                (start_at)
            VALUES
                (?)
            """;
    KeyHolder keyHolder = new GeneratedKeyHolder();
    jdbcTemplate.update(
            connection -> getPreparedStatement(reservationTime, connection, sql),
            keyHolder
    );
    long id = Objects.requireNonNull(keyHolder.getKey()).longValue();
    // return Objects.requireNonNull(keyHolder.getKey()).longValue();
    return new ReservationTime(id, reservationTime.getStartAt());
}
```
KeyHolder를 처음 사용해보다 보니 응용이 가능한 줄 몰랐습니다 😅 
`findById(Long id)` 메서드 형태로만 도메인 변환이 가능한 줄 알았습니다.
생각해보니 위와 같이 반환할 수 있네요!

ld 만 반환하는 경우도 생각해보았는데요.
Controller에서 요청 후 body에 도메인 전부가 필요하지 않는다면 , id만 반환해도 무방할거라 생각이 들었어요!

현재는 모든 도메인의 정보를 필요로 하니 제가 올린 코드로 수정해겠습니다!